sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -174,6 +174,7 @@ case class InsertIntoHadoopFsRelationCommand(
         qualifiedOutputPath
       }
 
+      val statsTrackers = Seq(basicWriteJobStatsTracker(hadoopConf))
```

Comment:
I'm a bit confused. Is this PR to extend the file source API, or data source v2? DS v2 uses `AppendDataExec` and its friends.

Reply:
The intention was to capture partition statistics during the writing and committing phases, and then report the results.  As an example, Iceberg does collect this information already, so the only necessary change is making this visible by asking for the aggregated results which are then posted to the bus.

The existing statistics that are gathered for V1 use `SQLMetric`, which is an `AccumulatorV2` constrained to a simple `Long` value.  This is why the change included an additional parameter which would allow for capturing the dynamic map of partitions to their stats.  I'll look to avoid this implementation detail in any interfaces.

The V2 updates make all of this optional, by adding a new interface for asking an implementing source to provide the aggregated results.

Reply:
Iceberg uses V2.  Iceberg was provided as an example where a source already supports collection of the partition metrics but doesn't make this information visible.